go/arrow/cdata/cdata_test_framework.go

```diff
@@ -277,6 +283,12 @@ func createCArr(arr arrow.Array) *CArrowArray {
 	carr.length = C.int64_t(arr.Len())
 	carr.null_count = C.int64_t(arr.NullN())
 	carr.offset = C.int64_t(arr.Data().Offset())
+	carr.release = (*[0]byte)(C.release_test_arr)
+
+	if !arrow.IsUnion(arr.DataType().ID()) && !internal.DefaultHasValidityBitmap(arr.DataType().ID()) {
```

Comment:
I meant more something along the lines of `len(buffers) > 0`. Why do we have to special-case this for unions and the validity bitmap? Or is this `if` clause even useful?

Reply:
Let me double check. I think there are still areas of the code which assume there is at least one element in the list of buffers (even if it's nil). For REE arrays (along with others) we initialize the buffer list with a single nil. I'll look through the code and see if it's safe enough to just let the slice of buffers be empty.